lib/kennel.rb

```diff
@@ -54,48 +54,88 @@ class << self
   self.err = $stderr
 
   class Engine
-    attr_accessor :strict_imports
+    def initialize(
+      update_generated: true,
+      show_plan: false,
+      require_confirm: nil,
+      update_datadog: false,
+      strict_imports: true
+    )
+      @update_generated = update_generated
+      @show_plan = show_plan
+      @require_confirm =
+        if require_confirm.nil?
+          default_require_confirm?
+        else
+          require_confirm
+        end
+      @update_datadog = update_datadog
+      @strict_imports = strict_imports
+    end
+
+    def run
+      if !update_generated? && !show_plan? && !update_datadog?
+        generated
+        return
+      end
+
+      if show_plan? || update_datadog?
+        # start generation and download in parallel to make planning faster
+        Utils.parallel([:generated, :definitions]) { |m| send m, plain: true }
+      end
+
+      if update_generated?
+        PartsSerializer.new(filter: filter).write(generated)
+      end
+
+      if show_plan? || update_datadog?
+        syncer # Instantiating the syncer calculates the plan
 
-    def initialize
-      @strict_imports = true
+        syncer.print_plan if show_plan?
+
+        if update_datadog?
+          syncer.update if update_datadog? && (!require_confirm? || syncer.confirm)
+        else
+          syncer.plan # i.e. get & return the already-calculated plan
+        end
+      end
     end
 
-    # start generation and download in parallel to make planning faster
-    def preload
-      Utils.parallel([:generated, :definitions]) { |m| send m, plain: true }
+    private
+
+    attr_reader :strict_imports
+
+    def default_require_confirm?
+      $stdin.tty? && $stdout.tty?
     end
 
-    def generate
-      parts = generated
-      PartsSerializer.new(filter: filter).write(parts) if ENV["STORE"] != "false" # quicker when debugging
-      parts
+    def update_generated?
+      @update_generated
     end
 
-    def plan
-      syncer.print_plan
-      syncer.plan
+    def show_plan?
+      @show_plan
     end
 
-    def update
-      syncer.print_plan
-      syncer.update if syncer.confirm
+    def require_confirm?
+      @require_confirm
     end
 
-    private
+    def update_datadog?
+      @update_datadog
+    end
```

Comment:
these accessors feel a bit awkward, would prefer the non `?` version

Reply:
It's a shame that ruby does make this easier. `attr_reader :foo, :bar?` would be very handy, if it existed.